c++/src/capnp/layout.c++

```diff
@@ -1708,7 +1721,7 @@ struct WireHelpers {
       BuilderArena* orphanArena = nullptr)) {
     // Allocate the space.
     word* ptr = allocate(ref, segment, capTable, roundBytesUpToWords(size),
-                         WirePointer::LIST, orphanArena);
+                         WirePointer::LIST, orphanArena, /*type=*/schema::Type::DATA);
```

Comment:
If the data blob's size is not evenly divisible by the word size (8 bytes), then there will be some bytes left over after the end of the data blob which will never be initialized here. E.g. if the application allocates a data blob of size 3, then there will be 5 bytes after it which are not part of the blob, and won't be part of the next allocation either; they'll just be skipped.

It is _critical_ that these bytes be zero'd here. Leaving them uninitialized means that the final message sent to the network will contain random uninitialized bytes, which may contain secrets. This is a major security bug.

In general I'm nervous about this change because of the possibility of bugs like this.